kubernetes-deployment/maintain-tool/repair-worker-node.sh

```diff
@@ -0,0 +1,37 @@
+#!/bin/bash
+
+# Copyright (c) Microsoft Corporation
+# All rights reserved.
+#
+# MIT License
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated
+# documentation files (the "Software"), to deal in the Software without restriction, including without limitation
+# the rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and
+# to permit persons to whom the Software is furnished to do so, subject to the following conditions:
+# The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING
+# BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+# NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM,
+# DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+docker stop kubelet
+docker rm kubelet
+
+for ID in `docker ps -a | awk "/k8s_/ {print\\$1}"`; do docker kill $ID; docker rm $ID ;  done
+
+# check etc/ exist or not.
+staticpod="src/etc"
+if [ -d "$staticpod" ]; then
+
+    cp -r src/etc /
+
+fi
+
+chmod u+x src/kubelet.sh
+./src/kubelet.sh
```

Comment:
>kubelet.sh [](start = 6, length = 10)

does it always work? for example, what if some service on this node requires to clean up the corresponding folder, otherwise the restart will fail?

Reply:
can you add some code to detect whether this is a worker node?

---
In reply to: [164973547](https://github.com/Microsoft/pai/pull/281#discussion_r164973547) [](ancestors = 164973547)